cmd/node-cache/main.go

```diff
@@ -176,28 +185,35 @@ func (c *cacheApp) parseAndValidateFlags() error {
 func (c *cacheApp) runChecks() {
 	for _, rule := range c.iptablesRules {
 		exists, err := c.iptables.EnsureRule(utiliptables.Prepend, rule.table, rule.chain, rule.args...)
-		if !exists {
-			if err != nil {
-				cache.teardownNetworking()
-				clog.Fatalf("Failed to add back non-existent rule %v", rule)
-			}
+		switch {
+		case exists:
+			// debug messages can be printed by including "debug" plugin in coreFile.
+			clog.Debugf("IP table rule (table %q, chain %q) already exists", rule.table, rule.chain)
+			continue
+		case err == nil:
 			clog.Infof("Added back nonexistent rule - %v", rule)
-		}
-		if err != nil {
-			clog.Errorf("Failed to check rule %v - %s", rule, err)
+			continue
+		// if we got here, either iptables check failed or adding rule back failed.
+		case isLockedErr(err):
+			clog.Infof("Failed to check/add back rule %v, due to xtables lock in use, retrying in %v", rule, c.params.interval)
+			setupErrCount.WithLabelValues("iptables_lock_error").Inc()
+		default:
+			clog.Errorf("Failed to add back non-existent rule %v - %s", rule, err)
+			setupErrCount.WithLabelValues("iptables_err").Inc()
```

Comment:
be consistent with err vs error? change to iptables_error or just remove _error from all of the labels (the metric is already has error in its name, seems kind of redundant.